cli/popper/commands/cmd_add.py

```diff
@@ -0,0 +1,74 @@
+import click
+import os
+import hcl
+import shutil
+import popper.utils as pu
+import popper.scm as scm
+from popper.cli import pass_context
+from distutils.dir_util import copy_tree
+
+
+@click.command('add', short_help='Import workflow from remote repo.')
+@click.argument('path', required=True)
+@pass_context
+def cli(ctx, path):
+    """Imports a workflow from a remote project to the current project
+    directory.
+    """
+    project_root = scm.get_root_folder()
+    if not pu.is_popperized(project_root):
+        pu.fail('Repository has not been popperized.')
+        return
+
+    parts = path.split('/')
+    parts = list(filter(None, parts))
+    if len(parts) < 3:
+        pu.fail(
+            'Required url format <url>/<user>/<repo>[/folder[/wf.workflow]]')
+
+    url, service, user, repo, _ = pu.parse('https://{}'.format(path))
+    if '@' in path:
+        branch = path.split('@')[-1]
+    else:
+        branch = 'master'
+
+    cloned_project_dir = os.path.join("/tmp", service, user, repo)
+    scm.clone(url, user, repo, os.path.dirname(
+        cloned_project_dir), branch
+    )
+
+    if len(parts) == 3:
```

Comment:
would it be possible to have this logic be part of the `Workflow` class? We can create a static `Workflow.import_from_repo` or something like that. Or alternatively a function in the `gha` module that implements this logic. The idea being that we can keep a separation between the logic of Popper and what is done in the CLI implementation.

Reply:
@ivotron Can you please explain me a bit why do we need a `Workflow.import_from_repo` when the cli logic is already seperated in `cmd_add.py` ?

Reply:
sure. The logic of adding a workflow to the current project is currently implemented in [this block](https://github.com/systemslab/popper/blob/0ec35ffc4b2754a4a2da1d70a47df766f48074c0/cli/popper/commands/cmd_add.py#L35-L74). If, for example, we implement a web service that exposes popper functionality via a REST interface, we would not be able to use this functionality unless we opened a shell and executed `popper add`. Instead, it is preferable to encapsulate all the logic of what Popper does in the `gha.py` module, as this can be used in contexts different than via a terminal.